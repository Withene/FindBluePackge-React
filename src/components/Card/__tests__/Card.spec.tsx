import { render, screen } from '@testing-library/react';
import { Card } from '../index';
import '@testing-library/jest-dom';

const a = {
  status: 'Objeto em tr√¢nsito - por favor aguarde',
  data: '17/08/2021',
  hora: '21:40',
  origem: 'Test',
  destino: 'Unidade de Tratamento - Brasilia-DF',
};
const b = {
  local: 'Unidade de Tratamento - Brasilia-DF',
};

describe('<Posts />', () => {
  it('should render posts With Origem', () => {
    render(<Card FirtsTitleProp={a.status} OrigemProp={a.origem} DestinoProp={a.destino} Time={a.hora} />);
    expect(screen.getByRole('heading', { name: /Destino: Unidade de Tratamento - Brasilia-DF/i })).toBeInTheDocument();
    expect(screen.getByText('Origem: Test')).toBeInTheDocument();
    expect(screen.getByText('Destino: Unidade de Tratamento - Brasilia-DF')).toBeInTheDocument();
    expect(screen.getByText('Horario: 21:40')).toBeInTheDocument();
  });

  it('should render posts With Local', () => {
    render(<Card FirtsTitleProp={a.status} LocalProp={b.local} Time={a.hora} />);
    expect(screen.getByRole('heading', { name: /Local: Unidade de Tratamento - Brasilia-DF/i })).toBeInTheDocument();
    expect(screen.getByText('Horario: 21:40')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <Card FirtsTitleProp={a.status} OrigemProp={a.origem} DestinoProp={a.destino} Time={a.hora} />,
    );
    expect(container).toMatchSnapshot();
  });
});
