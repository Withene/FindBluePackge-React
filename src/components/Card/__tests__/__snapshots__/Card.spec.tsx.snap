// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <div
    class="sc-bdfBQB rxjHx"
  >
    <div
      class="sc-gsTEea gAnMMC"
    >
      <h1
        class="sc-dlfnuX csyxjh"
      >
        Objeto em tr√¢nsito - por favor aguarde
      </h1>
      <h2
        class="sc-hKgJUU kQwUZX"
      >
        Destino: 
        Unidade de Tratamento - Brasilia-DF
         
      </h2>
      <div
        class="sc-gKseQn kDbISL"
      >
        <h2
          class="sc-eCstlR iVIYXC"
        >
          Origem: 
          Test
        </h2>
        <h2
          class="sc-jSgvzq gejCwp"
        >
          Horario: 
          21:40
        </h2>
      </div>
    </div>
  </div>
</div>
`;
